src/main/java/edu/harvard/iq/dataverse/harvest/server/xoai/DataverseXoaiItemRepository.java

```diff
@@ -0,0 +1,314 @@
+package edu.harvard.iq.dataverse.harvest.server.xoai;
+
+import io.gdcc.xoai.dataprovider.exceptions.handler.IdDoesNotExistException;
+import io.gdcc.xoai.dataprovider.filter.ScopedFilter;
+import io.gdcc.xoai.dataprovider.model.Item;
+import io.gdcc.xoai.dataprovider.model.ItemIdentifier;
+import io.gdcc.xoai.dataprovider.model.Set;
+import io.gdcc.xoai.dataprovider.model.MetadataFormat;
+import io.gdcc.xoai.dataprovider.repository.ItemRepository;
+import edu.harvard.iq.dataverse.Dataset;
+import edu.harvard.iq.dataverse.DatasetServiceBean;
+import edu.harvard.iq.dataverse.export.ExportException;
+import edu.harvard.iq.dataverse.export.ExportService;
+import edu.harvard.iq.dataverse.harvest.server.OAIRecord;
+import edu.harvard.iq.dataverse.harvest.server.OAIRecordServiceBean;
+import edu.harvard.iq.dataverse.util.StringUtil;
+import io.gdcc.xoai.dataprovider.exceptions.handler.HandlerException;
+import io.gdcc.xoai.dataprovider.exceptions.handler.NoMetadataFormatsException;
+import io.gdcc.xoai.dataprovider.repository.ResultsPage;
+import io.gdcc.xoai.model.oaipmh.ResumptionToken;
+import io.gdcc.xoai.model.oaipmh.results.record.Metadata;
+import io.gdcc.xoai.xml.EchoElement;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.time.Instant;
+import java.util.List;
+import java.util.logging.Logger;
+
+/**
+ *
+ * @author Leonid Andreev
+ * Implements an XOAI "Item Repository". Retrieves Dataverse "OAIRecords" 
+ * representing harvestable local datasets and translates them into 
+ * XOAI "items". 
+ */
+
+public class DataverseXoaiItemRepository implements ItemRepository {
+    private static Logger logger = Logger.getLogger("edu.harvard.iq.dataverse.harvest.server.xoai.DataverseXoaiItemRepository");
+    
+    private OAIRecordServiceBean recordService;
+    private DatasetServiceBean datasetService;
+    private String serverUrl; 
+
+    public DataverseXoaiItemRepository (OAIRecordServiceBean recordService, DatasetServiceBean datasetService, String serverUrl) {
+        super();
+        this.recordService = recordService;
+        this.datasetService = datasetService;
+        this.serverUrl = serverUrl; 
+    }
+    
+    private List<DataverseXoaiItem> list = new ArrayList<DataverseXoaiItem>();
+
+
+    @Override
+    public Item getItem(String identifier) throws IdDoesNotExistException {
+        // I'm assuming we don't want to use this version of getItem 
+        // that does not specify the requested metadata format, ever
+        // in our implementation - ? (L.A.)
+        throw new IdDoesNotExistException("Metadata Format is Required");
```

Comment:
**issue:** Please don't do this. There is exactly one valid use case for this: when a [`ListMetadataFormats`](http://www.openarchives.org/OAI/openarchivesprotocol.html#ListMetadataFormats) request is sent, it might carry an identifier to retrieve all available metadata formats for this specific item.

Please be aware that you extended the method signature: you provide a return type "Item", whereas the interface only request an "ItemIdentifier" (which is inherited by "Item", so this works).

The available metadata formats for the item are provided via the `Context`. It will default to add all formats the context knows about, but you can add more specific things if necessary.

I'd strongly advise against throwing the exception as is now.

Reply:
I didn't think/remember about this use case (ListMetadataFormats with the identifier parameter). Makes sense.
(Why is the method called getItem() though; and not getItemIdentifier(), for consistency with the other method?)

Reply:
Sure, will do.